### PR TITLE
feat(analytics): migrate quickview to EP

### PR DESCRIPTION
**Pull request by louis-bompart**
https://coveord.atlassian.net/browse/KIT-2984

- Rescope tests relative to quickviews.
- Do the migration

Closes https://github.com/coveo/analytics-migration-project/issues/28

### CODE REVIEW

File: packages/headless/src/controllers/insight/quickview/headless-insight-quickview.test.ts [hunk 1]
```diff
@@ -46,90 +57,44 @@ describe('Insight Quickview', () => {
...
+    const someHtmlRequestOptions: HtmlRequestOptions = {
+      uniqueId: 'some-id',
+      requestedOutputSize: 0,
+    };
 
-  [true, false].forEach((testValue) => {
-    it(`when the result #hasHtmlVersion is ${testValue} #state.resultHasPreview should be ${testValue}`, () => {
-      options.result = buildMockResult({hasHtmlVersion: testValue});
-      initQuickview();
+    mockedBuildCoreQuickview.mock.calls[0][2](state, someHtmlRequestOptions);
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/insight/quickview/headless-insight-quickview.test.ts:81]: At a glance, this is pretty cryptic to me and makes it dififcult to quickly grasp how the test actually works.

Looking at the Jest documentation and at the `buildCoreQuickview` function signature, I gather that we're calling the third argument of the first call made to the mocked `buildCoreQuickview`.

How about we do something like:

```suggestion
    const mockedBuildResultPreviewRequest = mockedBuildCoreQuickview.mock.calls[0][2];
    mockedBuildResultPreviewRequest(state, someHtmlRequestOptions);
```

in these kinds of tests?

This comments applies to all similar instances.

File: packages/headless/src/features/result-preview/result-preview-analytics-actions.test.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      engine = buildSearchEngine({
+        configuration: {
+          ...getSampleSearchEngineConfiguration(),
+        },
+      });
+    });
+
+    it('should call coveo.analytics.makeRecommendationOpen properly', async () => {
+      engine.dispatch(logDocumentQuickview(testResult));
+      await wait();
```
Comment (fbeaudoincoveo) [inline packages/headless/src/features/result-preview/result-preview-analytics-actions.test.ts:50]: Mmm.. why is this necessary?

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/headless/src/controllers/quickview/headless-quickview.test.ts [hunk 3]
```diff
@@ -1,32 +1,43 @@
...
+    engine = buildMockSearchEngine(preloadedState);
+  }
+
   function initQuickview() {
     quickview = buildQuickview(engine, {options});
   }
 
   beforeEach(() => {
-    engine = buildMockSearchAppEngine();
+    initEngine();
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/quickview/headless-quickview.test.ts:40]: I think you should also call `jest.resetAllMocks();` here

File: packages/headless/src/features/result-preview/result-preview-analytics-actions.test.ts (hunk 2 above)
Comment (louis-bompart) [reply packages/headless/src/features/result-preview/result-preview-analytics-actions.test.ts:50]: There are 'unbound' promises (i.e. out-of-reach promises) that we need to wait for.
This is a trick that allows us to flush the microtask queue to an extent.

File: packages/headless/src/controllers/insight/quickview/headless-insight-quickview.test.ts (hunk 1 above)
Comment (louis-bompart) [reply packages/headless/src/controllers/insight/quickview/headless-insight-quickview.test.ts:81]: > Looking at the Jest documentation and at the buildCoreQuickview function signature, I gather that we're calling the third argument of the first call made to the mocked buildCoreQuickview.

Yes

I'll work something out based on your suggestions (I don't want to use mock when it's not a mock, and the 3rd param is not a mock)

File: packages/headless/src/features/result-preview/result-preview-analytics-actions.test.ts (hunk 2 above)
Comment (fbeaudoincoveo) [reply packages/headless/src/features/result-preview/result-preview-analytics-actions.test.ts:50]: In the meantime, might be worth adding a little comment (maybe where we define the wait() function) to explain this cause it's not super obvious 😅

**Review by olamothe — APPROVED** [review]
